feat(tabs): update to use core tokens

mlogsdon18 opened the pull request:
## Description

This updates the Tabs component to use the new version of Spectrum tokens. 

## Related issue(s)
- #3255 

## Motivation and context

Part of our effort to migrate all Spectrum components to the new tokens.

## Known Design Changes
- Horizontal Tabs height has decreased by 2px
- Icon size went down by 2px in width (ex: 18px to 16px for Medium)

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply.  If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

mlogsdon18 commented:
@Westbrook this one is ready for your review! I believe that all of the VRT failures are expected and mostly due to some height changes in the tabs.

Westbrook reviewed (commented):
Be sure to add `tabs` to https://github.com/adobe/spectrum-web-components/blob/main/scripts/spectrum-tokens.js#L33...

Westbrook commented:
@dineshy87 as this is looking to address #3255, should we be adding borders in WHCM to Quiet Tabs, even though they do not have borders to begin with?

Westbrook commented:
@mlogsdon18 this one looks...not right? https://db7b4652508f151a03b04517a6e1a796--spectrum-web-components.netlify.app/review/#TabsStories/verticalQuietCompact.png

Westbrook commented:
In other contexts, we do not use "blue" for focus loops in WHCM: https://db7b4652508f151a03b04517a6e1a796--spectrum-web-components.netlify.app/review/#TopNavStories/autofocus.png Should this be updated?

dineshy87 commented:
> @dineshy87 as this is looking to address #3255, should we be adding borders in WHCM to Quiet Tabs, even though they do not have borders to begin with?

Yes. Adding border to quiet tabs as well in high contrast mode is a good thing. In my opinion.

mlogsdon18 commented:
@Westbrook I updated the Top Nav to use `canvastext` for the focus indicator color like we're doing in other places for WHCM. We'll need to do a follow-up to this PR once a new release of @spectrum-web-components/tabs goes out to update the Top Nav css to use the new variables, which I can't do now since we need a new release. 

I also updated the Quiet variant to have a bottom border in WHCM. Was there anything else that needed to be updated? I know you mentioned the vertical compact tabs which I am going to have Patrick do one more release to make the selected background area a bit larger to match the previous.

Westbrook commented on packages/top-nav/src/top-nav-item.css:
```diff
@@ -38,3 +41,9 @@ a:focus {
+
+@media (forced-colors: active) {
+    :host {
+        --highcontrast-tabs-focus-indicator-color: canvastext;
```
🔥

Westbrook commented on packages/top-nav/src/top-nav-item.css:
```diff
@@ -27,8 +27,11 @@ a:focus {
-        var(--spectrum-alias-border-color-key-focus)
+        --highcontrast-tabs-focus-indicator-color,
+        var(
+            --spectrum-tabs-m-focus-ring-color,
+            var(--spectrum-alias-border-color-key-focus)
```
Are these current to the new Tabs values?

Westbrook replied:
I mis-read! Internal to the repo, the dependencies always resolve to what's in the current branch, so you shouldn't need a new release to make this update. If it gives you trouble, that'll be important to know as it points to other issues.

mlogsdon18 replied:
Ok this has been updated! For some reason I tried this initially and it didn't work but looks like it's good now

najikahalsema reviewed (commented):
Looks good to me! 

The one thing I did notice is that it looks like the text is slightly high up in the element by one pixel (9 at the top, 11 at the bottom). Before, it was evenly 9 px on both sides. I measured it on [this screenshot](https://db7b4652508f151a03b04517a6e1a796--spectrum-web-components.netlify.app/review/#TabsVerticalSizesStories/m.png), as well as in a non-WHCM story in the storybook. It looks like it occurs in both horizontal and vertical cases... I don't know if this is something we want to bring back to the Spectrum team?

mlogsdon18 commented:
@najikahalsema Adding this in here too so it doesn't get lost in slack. 

Could you specify what you're measuring when you say "slightly high up in the element by one pixel". What is the "element" that you're basing the measurements on? I ask because depending on the size, according to the design, the spacing below the text is supposed to be one pixel greater than the spacing above (including the selection indicator)
![Screenshot 2023-06-30 at 2 34 23 PM](https://github.com/adobe/spectrum-web-components/assets/99203545/34bb7779-c396-4c00-925c-ba32c8f15c37)

najikahalsema commented:
> @najikahalsema Adding this in here too so it doesn't get lost in slack.
> 
> Could you specify what you're measuring when you say "slightly high up in the element by one pixel". What is the "element" that you're basing the measurements on? I ask because depending on the size, according to the design, the spacing below the text is supposed to be one pixel greater than the spacing above (including the selection indicator) ![Screenshot 2023-06-30 at 2 34 23 PM](https://user-images.githubusercontent.com/99203545/251181180-34bb7779-c396-4c00-925c-ba32c8f15c37.png)

I wasn't aware of this design spec, so if that's the case, it's good with me 👍

pfulton commented:
We've merged in the CSS PR, and graduated the release. The full release is: `@spectrum-css/tabs@4.0.0`

Westbrook reviewed (approved):
Thanks for bringing this one home for us @mlogsdon18! :shipit:
